vars/pushDockerImages.txt

```diff
@@ -0,0 +1,67 @@
+Publish docker images in the given docker registry. For such, it
+retags the existing docker images and publish them in the given
+docker namespace.
+
+It also allows to transform the version and customise private docker
+namespaces/registries.
+
+```
+  // Given the filebeat project, and its generated docker
+  // images for the 8.2.0-SNAPSHOT and 4 different variants
+  // then publish them to observability-ci
+  pushDockerImages(
+    secret: "my-secret",
+    registry: "my-registry",
+    arch: 'amd64',
+    version: '8.2.0',
+    snapshot: true,
+    imageName: 'filebeat',
+    variants: [
+      '' : 'beats',
+      '-ubi8' : 'beats',
+      '-cloud' : 'beats-ci',
+      '-complete' : 'beats',
+    ],
```

Comment:
What is the contract to define variant keys and values? I know it's something the consumer should know what to put here, but for having an overall reference, as I'm not able to infer any pattern with the current example

Reply:
The existing contract was already defined in the `packaging.groovy` as code:
- https://github.com/elastic/beats/blob/452d447372aac5449cf66e05c6e0eaaaf468dd66/.ci/packaging.groovy#L316-L326

This particular `variants` map provides:
- new variant name to be created and what targetNamespace.

Maybe the name could be changed to `variantsNamespaceMap` if it helps.

I still think this particular implementation could be much simpler if it's done by the build system itself. then from the CI point of view we only need to:

```
dockerLogin(...)
sh 'mage publishDockerImages'
```

For each beats.